mtags/src/main/scala-3/scala/meta/internal/pc/PcInlineValueProviderImpl.scala

```diff
@@ -0,0 +1,166 @@
+package scala.meta.internal.pc
+
+import scala.meta.internal.metals.CompilerOffsetParams
+import scala.meta.internal.metals.CompilerRangeParams
+import scala.meta.internal.mtags.MtagsEnrichments.*
+import scala.meta.internal.pc.AutoImports
+import scala.meta.internal.pc.AutoImports.SymbolIdent
+import scala.meta.internal.pc.InlineValueProvider.Errors
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.PresentationCompilerConfig
+
+import dotty.tools.dotc.ast.tpd.*
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags.*
+import dotty.tools.dotc.core.Symbols.NoSymbol
+import dotty.tools.dotc.core.Symbols.Symbol
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.util.SourcePosition
+import org.eclipse.{lsp4j as l}
+
+final class PcInlineValueProviderImpl(
+    val driver: InteractiveDriver,
+    val params: OffsetParams,
+    val config: PresentationCompilerConfig,
+) extends PcCollector[Occurence](driver, params)
+    with InlineValueProvider:
+
+  val text = params.text.toCharArray()
+
+  val position: l.Position = pos.toLsp.getStart()
+
+  override def collect(parent: Option[Tree])(
+      tree: Tree,
+      pos: SourcePosition,
+      sym: Option[Symbol],
+  ): Occurence =
+    val (adjustedPos, _) = adjust(pos)
+    Occurence(tree, parent, adjustedPos)
+
+  override def defAndRefs(): Either[String, (Definition, List[Reference])] =
+    val allOccurences = result()
+    for
+      definition <- allOccurences
+        .map(_.mkDefn)
+        .collectFirst { case Some(v) => v }
+        .toRight(Errors.didNotFindDefinition)
+      symbols = symbolsUsedInDefn(definition.tree.rhs)
+      references <- getReferencesToInline(definition, allOccurences, symbols)
+    yield
+      val (deleteDefinition, refsEdits) = references
+      val defEdit = Definition(
+        definition.tree.sourcePos.toLsp,
+        text
+          .slice(
+            definition.tree.rhs.sourcePos.start,
+            definition.tree.rhs.sourcePos.end,
+          )
+          .mkString,
+        RangeOffset(
+          definition.tree.sourcePos.start,
+          definition.tree.sourcePos.end,
+        ),
+        deleteDefinition,
+      )
+      (defEdit, refsEdits)
+    end for
+  end defAndRefs
+
+  private def symbolsUsedInDefn(
```

Comment:
Just an idea, maybe we can someday refactor PcCollector to allow passing it tree to travel instead of always traversing through the whole file, then we could use it here.